src/y-websocket.js

```diff
@@ -110,15 +111,20 @@ const setupWS = provider => {
       // log10(wsUnsuccessfulReconnects).
       // The idea is to increase reconnect timeout slowly and have no reconnect
       // timeout at the beginning (log(1) = 0)
-      setTimeout(setupWS, math.min(math.log10(provider.wsUnsuccessfulReconnects + 1) * reconnectTimeoutBase, maxReconnectTimeout), provider)
+      setTimeout(() => {
+        if (provider.reconnectPredicate(provider.wsUnsuccessfulReconnects)) {
```

Comment:
I think it would make sense to wrap the `setTimeout` inside this condition.

In case the predicate returns false, you should probably set `provider.wsconnecting = false; provider.shouldConnect = false; provider.wsUnsuccessfulReconnects = 0`. Then you can call `provider.connect()` again to try again.